### PR TITLE
Fix bug 915858 circular deps

### PR DESCRIPTION
**Pull request by i-garrison**
- split out new `media-libs/libmpg123` + fix masks for cpu flags in profiles
- adjust deps for `media-libs/libsndfile` to fix circular dependency
- while at it, adjust deps for `media-video/vlc` and `media-plugins/gst-plugins-mpg123`

### CODE REVIEW

**Comment by i-garrison** [comment]
Upstream PR https://github.com/madebr/mpg123/pull/11

**Comment by i-garrison** [comment]
> Upstream PR [madebr/mpg123#11](https://github.com/madebr/mpg123/pull/11)

Upstream suggests splitting out output modules instead of allowing building `mpg123` binary against installed libs  https://github.com/madebr/mpg123/pull/11#issuecomment-1798226267 - I do not think this will fit usual Gentoo split scheme of libs + binary package. Please chime in with ideas.

We can also just apply this to Gentoo tree and see if this package split can be improved later.

**Comment by i-garrison** [comment]
Rebased to recent master

File: media-sound/mpg123/files/mpg123-1.32.3-build-with-installed-libs.patch [hunk 1]
```diff
@@ -0,0 +1,251 @@
+build: allow building with installed libmpg123 libsyn123 libout123
```
Comment (thesamesam) [inline media-sound/mpg123/files/mpg123-1.32.3-build-with-installed-libs.patch:1]: Could you add a link to the Gentoo bug and upstream PR at the top of each of the patches?

Comment (i-garrison) [reply media-sound/mpg123/files/mpg123-1.32.3-build-with-installed-libs.patch:1]: Done!

File: profiles/arch/amd64/package.use.mask [hunk 2]
```diff
@@ -479,7 +479,7 @@ media-libs/sdl-gfx cpu_flags_x86_mmx
 media-libs/smpeg cpu_flags_x86_mmx
 
 # x86_64 opts are enabled with USE sse. so masking the rest
-media-sound/mpg123 cpu_flags_x86_mmx cpu_flags_x86_3dnow cpu_flags_x86_3dnowext
```
Comment (thesamesam) [inline profiles/arch/amd64/package.use.mask:482]: I think we need to duplicate this entry for the old mpg123 until new versions are stabled & old is cleaned up

File: media-video/vlc/vlc-3.0.20-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,517 @@
+# Copyright 2000-2023 Gentoo Authors
```
Comment (thesamesam) [inline media-video/vlc/vlc-3.0.20-r1.ebuild:1]: update *9999* too

File: profiles/arch/amd64/package.use.mask (hunk 2 above)
Comment (i-garrison) [reply profiles/arch/amd64/package.use.mask:482]: OK I restored original  masks in all 3 places

File: media-video/vlc/vlc-3.0.20-r1.ebuild (hunk 3 above)
Comment (i-garrison) [reply media-video/vlc/vlc-3.0.20-r1.ebuild:1]: Updated `vlc-3.0.9999` `vlc-9999` and `libsndfile-9999`

**Comment by thesamesam** [comment]
Thanks! Merged with `Bug:` tags on each commit to help people understand why.

**Pull request closed without merging**